nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -0,0 +1,446 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"strings"
+	"time"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func resourceNsxtManagerCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtManagerClusterCreate,
+		Read:   resourceNsxtManagerClusterRead,
+		Update: resourceNsxtManagerClusterUpdate,
+		Delete: resourceNsxtManagerClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"node": {
+				Type:        schema.TypeList,
+				Description: "Nodes in the cluster",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the cluster node",
+							Computed:    true,
+						},
+						"fqdn": {
+							Type:        schema.TypeString,
+							Description: "Fqdn cluster node",
+							Computed:    true,
+						},
+						"status": {
+							Type:        schema.TypeString,
+							Description: "Status of the cluster node",
+							Computed:    true,
+						},
+						"ip_address": {
+							Type:         schema.TypeString,
+							Description:  "IP address of the cluster node that will join the cluster of the host node",
+							ValidateFunc: validation.IsIPAddress,
+							Required:     true,
+						},
+						"password": {
+							Type:        schema.TypeString,
+							Description: "The authentication password for login",
+							Required:    true,
+							Sensitive:   true,
+						},
+						"username": {
+							Type:        schema.TypeString,
+							Description: "The username for login",
+							Required:    true,
+						},
+					},
+				},
+				Required: true,
+			},
+		},
+	}
+}
+
+type NsxClusterNode struct {
+	ID        string
+	IPAddress string
+	UserName  string
+	Password  string
+	Fqdn      string
+	Status    string
+}
+
+func getClusterNodesFromSchema(d *schema.ResourceData) []NsxClusterNode {
+	nodes := d.Get("node").([]interface{})
+	var clusterNodes []NsxClusterNode
+	for _, node := range nodes {
+		data := node.(map[string]interface{})
+		id := data["id"].(string)
+		ipAddress := data["ip_address"].(string)
+		userName := data["username"].(string)
+		password := data["password"].(string)
+		fqdn := data["fqdn"].(string)
+		status := data["status"].(string)
+		nodeObj := NsxClusterNode{
+			ID:        id,
+			IPAddress: ipAddress,
+			UserName:  userName,
+			Password:  password,
+			Fqdn:      fqdn,
```

Comment:
We don't need to include computed fields when reading from schema. Lets include the required minimum, I believe it would improve code readability.